src/pages/index.js

```diff
@@ -4,53 +4,136 @@ import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import Section from "../components/Section.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/Api.js";
 import "./index.css";
 import jacquesCousteau from "../images/jacques-cousteau.jpg";
 import logoImage from "../images/logo.svg";
 import {
   settings,
-  initialCards,
   profileEditButton,
   cardAddButton,
+  avatarEditButton,
 } from "../utils/constants.js";
 
 //Functions
 function handleEditProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profilePopupForm.close();
+  profilePopupForm.setLoading(true);
+  api
+    .editProfile({ name: data.name, description: data.description })
+    .then((res) => {
+      userInfo.setUserInfo({ name: res.name, description: res.about });
+      profilePopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  profilePopupForm.setLoading(false);
   return;
 }
 
+function handleAvatarFormSubmit(data) {
+  avatarPopupForm.setLoading(true);
+  api
+    .updateAvatar(data.image)
+    .then((res) => {
+      avatarPopupForm.close();
+      avatarPopupForm.reset();
+      userInfo.setAvatar(data.image);
+    })
+    .catch((err) => {
+      console.error("Error saving avatar:", err);
+    })
+    .finally(avatarPopupForm.setLoading(false));
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 

**Please, correct it everywhere in the code**

Reply:
I've made sure to pass arrow functions within the finally blocks to ensure it would run correctly. Also removed the extra "res" from the argument of a "then" block that did not use it.